pkg/liqonet/iptables/iptables.go

```diff
@@ -481,7 +532,227 @@ func (h IPTHandler) EnsureForwardExtRules(tep *netv1alpha1.TunnelEndpoint) error
+		// Create IP set
+		setName := getClusterIPSetForOffloadedPods(clusterID)
+		ipset, err := ipSetHandler.CreateSet(setName, "")
+		if err != nil {
+			klog.Infof("Error while creating IP set %q: %w", setName, err)
+			return nil, err
+		}
+
+		// Clear IP set (just in case it already existed)
+		if err := ipSetHandler.FlushSet(ipset.Name); err != nil {
+			klog.Infof("Error while deleting all entries from IP set %q: %w", setName, err)
+			return nil, err
+		}
+
+		if len(ips) > 0 {
+			for _, podIP := range ips {
+				// Add pod's IP entry to IP set
+				if err := ipSetHandler.AddEntry(podIP, ipset); err != nil {
+					klog.Infof("Error while adding entry %q to IP set %q: %w", podIP, ipset.Name, err)
+					return nil, err
+				}
+			}
+
+			// Add match-set rule
+			rulesPerCluster[clusterID] = append(
+				rulesPerCluster[clusterID],
+				IPTableRule{
+					"-m", "comment", "--comment",
+					// WARNING: Never use double-quotes inside the comment, otherwise IpTableRule parser will fail
+					fmt.Sprintf("Allows traffic from '%s' only to pods offloaded by that remote cluster", clusterID),
+					"-m", setModule,
+					"--match-set", ipset.Name, "dst",
+					"-j", ACCEPT})
+		}
+	}
+
+	return rulesPerCluster, nil
+}
+
+// buildRulesPerClusterForEndpointslicesReflected builds rules allowing traffic towards endpoints of local services reflected on other clusters.
+func buildRulesPerClusterForEndpointslicesReflected(
+	endpointslicesInfo *sync.Map,
+	ipSetHandler *liqoipset.IPSHandler,
+) (map[string][]IPTableRule, error) {
+	// Map of Pod IPs per cluster
+	endpointSetsPerCluster := map[string]map[string][]string{}
+
+	// Populate Pod IPs per cluster
+	endpointslicesInfo.Range(func(key, value any) bool {
+		namespacedName := key.(types.NamespacedName)
+		endpointsInfo := value.(map[string]EndpointInfo)
+		for _, endpointInfo := range endpointsInfo {
+			for _, clusterID := range endpointInfo.SrcClusterIDs {
+				if _, ok := endpointSetsPerCluster[clusterID]; !ok {
+					endpointSetsPerCluster[clusterID] = map[string][]string{}
+				}
+				if _, ok := endpointSetsPerCluster[clusterID][namespacedName.String()]; !ok {
+					endpointSetsPerCluster[clusterID][namespacedName.String()] = []string{}
+				}
+				if !endpointInfo.Deleting {
+					endpointSetsPerCluster[clusterID][namespacedName.String()] = append(
+						endpointSetsPerCluster[clusterID][namespacedName.String()], endpointInfo.Address)
+				}
+			}
+		}
+		return true
+	})
+
+	// Map of IPTables rules and IP sets per cluster
+	rulesPerCluster := map[string][]IPTableRule{}
+
+	// Populate IP set per endpointslice and cluster, and  IPTables rules per cluster
+	for clusterID, endpointsSets := range endpointSetsPerCluster {
+		rulesPerCluster[clusterID] = []IPTableRule{}
+		for namespacedName, endpointSet := range endpointsSets {
+			setName := fmt.Sprintf("%s-%s", strings.ToUpper(strings.Split(namespacedName, "/")[1]), strings.Split(clusterID, "-")[0])
```

Comment:
Plaese check that the access to the string split do not panics. And make the split only once

Reply:
I can't split once because are two different strings.
For the second one I don't check the access, the string split with index 0 is guaranteed by Split function.

Reply:
ok, check the other